holidays/ical.py

```diff
@@ -0,0 +1,184 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+import uuid
+from datetime import datetime, timedelta, timezone
+
+from holidays.version import __version__
+
+# iCal-specific constants
+ICAL_CHARSET = "utf-8"
```

Comment:
Since utf-8 is the default encoding for both [iCal](https://datatracker.ietf.org/doc/html/rfc5545.html#section-3.1.4) and str.encode, I think we can just omit it.